MiniGamesBox Classic/src/main/java/plugily/projects/minigamesbox/classic/arena/PluginArenaUtils.java

```diff
@@ -70,50 +72,49 @@ public static void hidePlayersOutsideTheGame(Player player, PluginArena arena) {
     }
   }
 
-  public static void preparePlayerForGame(
-      PluginArena arena, Player player, Location location, boolean spectator) {
-    User user = plugin.getUserManager().getUser(player);
-    if(plugin.getConfigPreferences().getOption("INVENTORY_MANAGER")) {
-      InventorySerializer.saveInventoryToFile(plugin, player);
-    }
-    VersionUtils.teleport(player, location);
-    player.getInventory().clear();
-    player
-        .getActivePotionEffects()
-        .forEach(potionEffect -> player.removePotionEffect(potionEffect.getType()));
-    VersionUtils.setMaxHealth(player, VersionUtils.getMaxHealth(player));
-    player.setHealth(VersionUtils.getMaxHealth(player));
-    player.setFoodLevel(20);
-    player.setGameMode(GameMode.SURVIVAL);
-    player
-        .getInventory()
-        .setArmorContents(
-            new ItemStack[]{
-                new ItemStack(Material.AIR),
-                new ItemStack(Material.AIR),
-                new ItemStack(Material.AIR),
-                new ItemStack(Material.AIR)
-            });
-    player.setExp(1);
-    player.setLevel(0);
-    player.setWalkSpeed(0.2f);
-    player.setFlySpeed(0.1f);
+  public static CompletableFuture<Void> preparePlayerForGame(PluginArena arena, Player player, Location location, boolean spectator) {
+    return VersionUtils.teleport(player, location).thenAccept(bol -> {
+      if(plugin.getConfigPreferences().getOption("INVENTORY_MANAGER")) {
+        InventorySerializer.saveInventoryToFile(plugin, player);
```

Comment:
Preparing a player for game would require to save the inventory before teleportation so if in world a it needs to save the inv from world a and then teleport to world b. so the old behavoir for that was fine, as only the recreation was faulty

Reply:
Yea, but what if a server owner having a per-world inventory, which means if for example worldB have different inventory than worldA and the game is in worldB then the player's inventory from worldB will be lost if the player join to a game.

Reply:
thats correct but in general the user would get the inventory back which he got before the game started so we don't care about the world change which is made and will cause other plugins to change the inv. on teleport to end location you will get the inventory back you got before you started the game, in general that means its the same as before. The real issue comes up now if the end location is in another world, so the player gets a overwritten inventory, that can be resolved by just disabling the inventory option, so the inventory won't be handled by us in any way, which should also solve the issue

Reply:
In this case idk 🤷‍♂️